posts/2024-11-10-investigating-spirv-for-the-shader-system-part-2.md

```diff
@@ -0,0 +1,252 @@
+---
+title: "Investigating SPIR-V for the shader system - Part 2"
+author: youness
+popular: false
+image: /images/spir.png
+tags: ['.NET', 'Shaders']
+---
+This second part is going to dive deeper in how the current SDSL compiler works and how we are improving on it for the SPIR-V compiler. This will be a sort of personal log book of my research on the subject.
+
+---
+
+Table of Contents:
+
+[[TOC]]
+
+In the last blog post we've briefly went through the why SPIR-V was an interesting shader format for our new shader compiler. After a year of research on this subject we're coming back with ideas and implementations !
+
+### Preface
+
+Rewriting the shader system is a complicated tasks in itself. As you may already know, Stride was initially a commercial product that was open sourced not long ago, and some of the original devs left. This has led to a difficulty to gather information on the inner working of the engine.
+
+Fortunately we still have some of the original team members helping and answering questions, and the source code is well written and easy to understand, this has helped a lot of contributors to be able to fix issues quite fast.
+
+As for me, I started this project with no prior knowledge on how compiler works and a very faint idea on how shader works. I've learned some things and trying my best, so if you have any ideas, improvements and criticism, I'll be happy to discuss it on our Discord server or in a Github discussion :D
+
+### How it works currently
+
+There are two components to SDSL :
```

Comment:
The colon (":") should be placed directly after the word, with no space in between. This can be fixed throughout the entire article. Additionally, punctuation marks like "!" should also be positioned right after the last word. I hope it makes sense 🙂

Reply:
Yes that's true, i was still thinking in french while writing 😆